### PR TITLE
Added hit animations, updated lighting & added InteliJ to gitignore

### PR DESCRIPTION
**Pull request by LemonLake**
I hope I did this right ._. pardon the IntelliJ typo

### CODE REVIEW

**Comment by itdelatrisu** [comment]
Thanks for taking care of this!  There are a couple of bugs:
* The hit circles are being drawn for slider tick results.
* In osu!, the first and last circles in a slider get animated after completing the slider; in your implementation, the first circle is animated immediately.
osu! also has a kind of bouncy effect when you hit a circle, and it feels a little off seeing just a linear size increase.

There's a copyright issue, as well -- peppy has copyright on all the default osu! skin images, so I can't actually include the `lighting.png` image in this repository.  Users will have to supply it themselves.

**Comment by LemonLake** [comment]
This is the lighting texture I use in my own skin, so this one should be fine. I'll get to work on those bugs right away

**Comment by LemonLake** [comment]
I'm not quite sure what you mean by bounciness, but it definitely does look like there is some easing (60fps at 30% speed)

**Comment by Bigpet** [comment]
@LemonLake just implement whatever you need on top of the current slider system (something almost exactly like it will still exist as a fallback anyway), I'll rebase however necessary.

**Comment by LemonLake** [comment]
New bug: Geki and katu hit results no longer appear. Know the solution, just looking for the fix

File: src/itdelatrisu/opsu/objects/HitResultType.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package itdelatrisu.opsu.objects;
+
+public enum HitResultType {
+    CIRCLE,
+    SLIDERSTART,
```
Comment (itdelatrisu) [inline src/itdelatrisu/opsu/objects/HitResultType.java]: `SLIDERSTART` is never used.

File: src/itdelatrisu/opsu/GameData.java [hunk 2]
```diff
@@ -1152,11 +1244,12 @@ public void hitResult(int time, int result, float x, float y, Color color,
 		case HIT_MISS:
 			hitValue = 0;
 			changeHealth(-10f);
+			changeHealth(-10f);
```
Comment (itdelatrisu) [inline src/itdelatrisu/opsu/GameData.java]: Please explain?

File: src/itdelatrisu/opsu/Utils.java [hunk 3]
```diff
@@ -607,4 +607,14 @@ else if (seconds < 3600)
 		else
 			return String.format("%02d:%02d:%02d", seconds / 3600, (seconds / 60) % 60, seconds % 60);
 	}
+
```
Comment (itdelatrisu) [inline src/itdelatrisu/opsu/Utils.java]: Could you add Javadocs for these?

**Comment by itdelatrisu** [comment]
This looks really great now -- thanks again!  I added a few comments in the diff, but visually this seems perfect.  Are you ready for me to merge it?

File: src/itdelatrisu/opsu/objects/HitResultType.java (hunk 1 above)
Comment (LemonLake) [inline src/itdelatrisu/opsu/objects/HitResultType.java]: That was going to be used, but I never actually used it. I left it in there in the event that anyone in the future wanted to use it.

File: src/itdelatrisu/opsu/GameData.java (hunk 2 above)
Comment (LemonLake) [inline src/itdelatrisu/opsu/GameData.java]: I have no idea how that happened.

File: src/itdelatrisu/opsu/Utils.java (hunk 3 above)
Comment (LemonLake) [inline src/itdelatrisu/opsu/Utils.java]: Will do.

**Comment by LemonLake** [comment]
I'll just fix those and then it will be ready!

**Comment by LemonLake** [comment]
Go ahead and merge!

**Pull request merged**